docs/src/content/en/reference/templates.mdx

```diff
@@ -0,0 +1,222 @@
+---
+title: "Templates Reference"
+description: "Complete guide to creating, using, and contributing Mastra templates"
+---
+
+import { FileTree, Tabs, Callout } from 'nextra/components'
+
+This reference provides comprehensive information about Mastra templates, including how to use existing templates, create your own, and contribute to the community ecosystem.
+
+## Overview
+
+Mastra templates are pre-built project structures that demonstrate specific use cases and patterns. They provide:
+
+- **Working examples** - Complete, functional Mastra applications
+- **Best practices** - Proper project structure and coding conventions
+- **Educational resources** - Learn Mastra patterns through real implementations
+- **Quick starts** - Bootstrap projects faster than building from scratch
+
+## Using Templates
+
+### Installation
+
+Install a template using the `create-mastra` command:
+
+```bash copy
+npx create-mastra@latest --template template-name
+```
+
+This creates a complete project with all necessary code and configuration.
+
+### Setup Process
+
+After installation:
+
+1. **Navigate to project directory**:
+
+   ```bash copy
+   cd your-project-name
+   ```
+
+2. **Configure environment variables**:
+
+   ```bash copy
+   cp .env.example .env
+   ```
+
+   Edit `.env` with required API keys as documented in the template's README.
+
+3. **Install dependencies** (if not done automatically):
+
+   ```bash copy
+   npm install
+   ```
+
+4. **Start development server**:
+
+   ```bash copy
+   npm run dev
+   ```
+
+### Template Structure
+
+All templates follow this standardized structure:
+
+<FileTree>
+  <FileTree.Folder name="template-name" defaultOpen>
+    <FileTree.Folder name="src" defaultOpen>
+      <FileTree.Folder name="mastra" defaultOpen>
+        <FileTree.Folder name="agents">
+          <FileTree.File name="example-agent.ts" />
+        </FileTree.Folder>
+        <FileTree.Folder name="tools">
+          <FileTree.File name="custom-tool.ts" />
```

Comment:
Is there a reason this file is called `custom-tool` and the agent and workflow are `example-`?